src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -0,0 +1,160 @@
+   * list is sorted in reverse order by commit date. This will ensure that the commit list will hold
+   * data from latest commit to the first commit. This sorted list will then be stored in a cache
+   * ({@link LRUMap}).
+   * 
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return List of {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  private List<GHCommit> commitsFor(GitHubProject project) throws IOException {
+    Optional<List<GHCommit>> cachedCommits = commitsCache.get(project);
+    if (cachedCommits.isPresent()) {
+      return cachedCommits.get();
+    }
+
+    List<GHCommit> commits = new ArrayList<GHCommit>(repositoryFor(project).listCommits().toList());
+    commits.sort(BY_COMMIT_DATE.reversed());
+    commitsCache.put(project, commits, tomorrow());
+    return commits;
+  }
+
+  /**
+   * Gets the GitHub project repository. This repository will then be stored in a cache
+   * ({@link LRUMap}).
+   * 
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return {@link GHRepository} with the project information.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public GHRepository repositoryFor(GitHubProject project) throws IOException {
+    Optional<GHRepository> cachedRepository = repositoryCache.get(project);
+    if (cachedRepository.isPresent()) {
+      return cachedRepository.get();
+    }
+
+    GHRepository repository = github.getRepository(project.path());
+    repositoryCache.put(project, repository, tomorrow());
+    return repository;
+  }
+
+  /**
+   * Gets the first commit of the repository.
+   * 
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return {@link GHCommit} with the project information.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public Optional<GHCommit> firstCommitFor(GitHubProject project) throws IOException {
+    List<GHCommit> commits = commitsFor(project);
+    if (commits.isEmpty()) {
+      return Optional.empty();
+    }
+    return Optional.ofNullable(commits.get(commits.size() - 1));
+  }
+
+  /**
+   * Get a list of commits which comes after a specific date.
+   * 
+   * @param date which is used as boundary.
+   * @param project of type {@link GitHubProject}, which holds the project information.
+   * @return sub-list of type {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public List<GHCommit> commitsAfter(Date date, GitHubProject project)
+      throws IOException {
+    List<GHCommit> subList = new ArrayList<GHCommit>();
+
+    for (GHCommit commit : commitsFor(project)) {
+      if (commit.getCommitDate().before(date)) {
+        break;
+      }
+      subList.add(commit);
+    }
+
+    return Collections.unmodifiableList(subList);
```

Comment:
`unmodifiableList` doesn't seem to be necessary here since you create a new list `subList`.